plugins/modules/ise_radius_integration_workflow_manager.py

```diff
@@ -0,0 +1,1466 @@
+        if server_type not in ["ISE", "AAA", None]:
+            self.msg = "server_type should either be ISE or AAA but not {0}.".format(server_type)
+            self.status = "failed"
+            return self
+
+        if server_type == "ISE":
+            auth_server.update({"isIseEnabled": True})
+        else:
+            auth_server.update({"isIseEnabled": False})
+
+        auth_server.update({"ipAddress": auth_policy_server.get("server_ip_address")})
+
+        shared_secret = auth_policy_server.get("shared_secret")
+        if not (shared_secret or auth_server_exists):
+            self.msg = "shared_secret is mandatory parameter"
+            self.status = "failed"
+            return self
+
+        if not (4 <= len(shared_secret) <= 10) or shared_secret.isspace():
+            self.msg = "shared_secret should character should be between 4 to 100."
+            self.status = "failed"
+            return self
+
+        auth_server.update({"sharedSecret": shared_secret})
+
+        protocol = auth_policy_server.get("protocol")
+        if protocol not in ["RADIUS", "TACACS", "RADIUS_TACACS", None]:
+            self.msg = "protocol should either be ['RADIUS', 'TACACS', 'RADIUS_TACACS']." + \
+                       "It should not be {0}".format(protocol)
+            self.status = "failed"
+            return self
+
+        if protocol is not None:
+            auth_server.update({"protocol": protocol})
+        else:
+            auth_server.update({"protocol": "RADIUS"})
+
+        encryption_scheme = str(auth_policy_server.get("encryption_scheme"))
+        if encryption_scheme not in ["KEYWRAP", "RADSEC", None]:
+            self.msg = "encryption_scheme should be in ['KEYWRAP', 'RADSEC']. " + \
+                       "It should not be {0}.".format(encryption_scheme)
+            self.status = "failed"
+            return self
+
+        if encryption_scheme:
+            auth_server.update({"encryptionScheme": encryption_scheme})
+
+        if encryption_scheme == "KEYWRAP":
+            message_key = str(auth_policy_server.get("message_key"))
+            if not message_key:
+                self.msg = "message_key should not be empty if encryption_scheme is 'KEYWRAP'."
+                self.status = "failed"
+                return self
+
+            if len(message_key) != 20:
+                self.msg = "message_key should be exactly 20 character."
+                self.status = "failed"
+                return self
+
+            auth_server.update({"messageKey": message_key})
+
+            encryption_key = auth_policy_server.get("encryption_key")
+            if not encryption_key:
+                self.msg = "encryption_key should not be empty if encryption_scheme is 'KEYWRAP'."
+                self.status = "failed"
+                return self
+
+            if len(encryption_key) != 16:
+                self.msg = "encryption_key should be exactly 16 characters."
```

Comment:
When ASCII format is selected, Encryption Key may contain alphanumeric and special characters. Key must be 16 char long. When HEXADECIMAL format is selected, Encryption Key may contain hexadecimal pairs ranging from 21 to 7E. Key must be 32 char long.